test/modules/encoders/test_clip_text_encoder.py

```diff
@@ -20,85 +20,136 @@ def start(self):
+                embedding_dim=embedding_dim,
+                use_clip_init=use_clip_init,
+                context_length=context_length,
             )
 
-        return build_encoder, build_text, context_length
+        return build_encoder, build_text
 
     def test_initialization(self, start):
-        build_encoder, build_text, context_length = start
-        text_length = min(context_length, 3)
-        text = build_text(text_length)
+        """Test whether the `use_clip_init` parameter changes the initialized parameters."""
+        build_encoder, _ = start
 
-        # First, initialize without CLIP parameters
+        # First, initialize without CLIP parameters and get the standard deviation of token embedding weights
         text_encoder = build_encoder(use_clip_init=False)
         assert isinstance(text_encoder, torch.nn.Module)
-
-        actual_no_init = text_encoder(text)
-        expected_no_init = torch.Tensor(
-            [[-0.3195, 0.1647, -0.2621, 0.4387], [-0.3354, 0.2859, -0.6949, 0.2047]]
-        )
-        assert_expected(
-            actual=actual_no_init, expected=expected_no_init, rtol=0, atol=1e-4
-        )
-
-        # Standard deviation of token embedding weights should be different without vs with CLIP params
         actual_std_no_init = torch.std(
             text_encoder.encoder.token_embedding.weight
         ).item()
-        expected_std_clip_init = CLIPTextEncoder.TOKEN_EMBEDDING_INIT_STD
-        assert abs(actual_std_no_init - expected_std_clip_init) > 1e-4
 
-        # Then test the default CLIP initialization
+        # Get the standard deviation with CLIP params
         text_encoder.initialize_parameters()
+        actual_std_clip_init = torch.std(
+            text_encoder.encoder.token_embedding.weight
+        ).item()
 
-        actual_clip_init = text_encoder(text)
-        expected_clip_init = torch.Tensor(
-            [[-1.9412, -0.8044, -0.7616, -0.3185], [0.2408, -0.3241, -0.2544, -0.3608]]
+        assert abs(actual_std_clip_init - actual_std_no_init) > 1e-4
+
+    def test_clip_parameters(self, start):
+        build_encoder, _ = start
+        # use larger embedding size for stability in std
+        text_encoder = build_encoder(embedding_dim=50)
+
+        assert_expected(
+            actual=torch.std(text_encoder.encoder.token_embedding.weight).item(),
+            expected=0.02,
+            rtol=0,
+            atol=1e-4,
         )
         assert_expected(
-            actual=actual_clip_init, expected=expected_clip_init, rtol=0, atol=1e-4
+            actual=torch.std(
+                text_encoder.encoder.positional_embedding.embedding.weight
+            ).item(),
+            expected=0.01,
+            rtol=0,
+            atol=1e-4,
         )
 
-        actual_std_clip_init = torch.std(
-            text_encoder.encoder.token_embedding.weight
-        ).item()
+        proj_std = (text_encoder.width**-0.5) * (
```

Comment:
Compute ground truths independently to avoid copying the logic from the code you are testing.